src/Blubrry/REST/Resource/Social.php

```diff
@@ -49,6 +68,85 @@ public function updateListing($program_keyword, $body) {
     public function getSocial($program_keyword, $params) {
         $path = '/2/social/' . $program_keyword . '/get-social-options.json';
 
+        $social_options = [];
+        $social_options['social-options'] = [];
+
+        $social_types = ['twitter', 'youtube', 'facebook'];
+        $required_fields = ['social-id', 'social-image', 'social-title'];
+        $attributes = [
+            'html' => [
+                'required' => [],
+                'optional' => ['content'],
+            ],
+            'input-text' => [
+                'required' => ['label', 'name'],
+                'optional' => ['placeholder', 'help-text', 'rows', 'maxlength', 'value'],
+            ],
+            'input-checkbox' => [
+                'required' => [],
+                'optional' => ['label', 'checked', 'name', 'value'],
+            ],
+            'input-radio' => [
+                'required' => [],
+                'optional' => ['label', 'checked', 'name', 'value'],
+            ],
+        ];
+
+        foreach ($social_types as $type) {
+            foreach ($params[$type] as $item) {
+                $SocialOption = [];
+
+                foreach ($required_fields as $field) {
+                    if (!empty($item[$field])) {
+                        $socialOption[$field] = $item[$field];
+                    } else {
+                        return false;
+                    }
+                }
+                
+                $SocialOption['social-type'] = $type;
+                $SocialOption['form-data'] = [];
+
+                if (is_array($item['form-data'])) {
+                    if (empty($item['form-data']['field-type'])) {
+                        return false;
+                    }
+
+                    if (empty($item['form-data']['field-order'])) {
+                        return false;
+                    }
+
+                    $type = $item['form-data']['field-type'];
+                    $order = $item['form-data']['field-order'];
+
+                    foreach ($attributes as $attr) {
+                        if ($attr == $type) {
+                            foreach ($attr['required'] as $req) {
+                                if (empty($item['form-data'][$req])) {
+                                    return false;
+                                } else {
+                                    $SocialOption['form-data'][$req] = $item['form-data'][$req];
+                                }
+                            }
+
+                            foreach ($attr['optional'] as $opt) {
+                                if (empty($item['form-data'][$opt])) {
```

Comment:
@lucashillebrandt I would verify for non-empty strings here, so you can simplify and get rid of the continue.

```php
if (!empty($item['form-data'][$opt])) {
    $SocialOption['form-data'][$opt] = $item['form-data'][$opt];
}
```